src/QuickMarcEditor/QuickMarcEditor.js

```diff
@@ -461,17 +464,23 @@ const QuickMarcEditor = ({
   };
 
   const cancelDeleteFields = () => {
-    setIsDeleteModalOpened(false);
-
     if (deletedRecords.length) {
       restoreDeletedRecords();
     } else {
       reset();
     }
+
+    setIsDeleteModalOpened(false);
+    setTimeout(() => {
+      focusLastFocusedInput();
+    });
   };
 
   const cancelRemoveLinking = () => {
     setIsUnlinkRecordsModalOpen(false);
+    setTimeout(() => {
+      focusLastFocusedInput();
+    });
```

Comment:
1. It looks like there is a need to close a modal first and then invoke the `focusLastFocusedInput`. Is it possible?
2. Where should the focus be after a field is linked/unlinked?

Reply:
I guess remove linking is not applicable for current story if it only happens instantly when a record has been opened for deriving. I'll remove this code because it's not needed.
But generally a modal will close first, and a last focused element will be re-focused second:

https://github.com/user-attachments/assets/aca85a69-4374-4bb8-b4cd-24d1f40a1132

Reply:
> But generally a modal will close first, and a last focused element will be re-focused second:

I meant `setTimeout` is used to trigger the focus after modal is closed (state is `false`) and whether it is possible to do it without `setTimeout`, just wait until modal state changes to `false`, but I think it will require a lot more changes.

Reply:
@Dmytro-Melnyshyn oh I see. it's not because we're waiting for the modal to close, but we need the form to finish all updates. If we set focus right away it'll be gone when the form updates. And there's no way to know when it does so we have to use `setTimeout` here